proposal.md

```diff
@@ -0,0 +1,1451 @@
+```
+
+The `ErrUnknownName` error was added to support common use-cases
+wanting to distinguish this particular condition (see #29035).
+
+The following behavior changes were made to marshal and unmarshal:
+
+* There is newely added support for the `strictcase` option to provide
+  a better migration path between users of both v1 and v2.
+
+* Specifying the `string` tag option now rejects unmarshaling
+  from a JSON number and only permits unmarshaling from a JSON string.
+  This exactly matches the behavior of v1.
+
+* When unmarshaling, a floating-point overflow results in an error.
+  This exactly matches the behavior of v1.
+
+* Serialization now supports embedded fields of unexported struct types
+  with exported fields.
+  This exactly matches the behavior of v1.
+
+### Package "encoding/json"
+
+Some options for legacy v1 support were renamed or
+had similar options folded together.
+```diff
+- func RejectFloatOverflow(v bool) Options
+
+- func IgnoreStructErrors(v bool) Options
+- func ReportLegacyErrorValues(v bool) Options
++ func ReportErrorsWithLegacySemantics(bool)
+
+- func SkipUnaddressableMethods(v bool) Options
++ func CallMethodsWithLegacySemantics(bool)
+
+- func FormatByteArrayAsArray(v bool) Options
++ func FormatBytesWithLegacySemantics(bool)
+
+- func FormatTimeDurationAsNanosecond(v bool) Options
++ func FormatTimeWithLegacySemantics(bool)
+
++ func EscapeInvalidUTF8(bool)
+```
+In general, many options were renamed with a `WithLegacySemantics` suffix
+because they convered a multitude of behavior differences that could
+not be adequently described with a concise name.
+
+The `RejectFloatOverflow` option was removed because v2 now rejects
+floating-point overflows just like v1.
+
+The `EscapeInvalidUTF8` option was added in order to support a
+behavior difference that was discovered while implementing
+v1 support in terms of v2.
+
+The `Number` type implements `MarshalerTo` and `UnmarshalerFrom` for
+better compatibility with v2.
+```diff
++ func (Number) MarshalJSONTo(*jsontext.Encoder, jsonopts.Options) error
++ func (*Number) UnmarshalJSONFrom(*jsontext.Decoder, jsonopts.Options) error
+```
+
+The `UnmarshalTypeError` type now supports error wrapping.
+```diff
+  type UnmarshalTypeError struct {
+  	...
++ 	Err error
+  }
+
++ func (*UnmarshalTypeError) Unwrap() error
+```
+
+## Proposed implementation
+
+This proposal has been implemented by the [`github.com/go-json-experiment/json`](https://pkg.go.dev/github.com/go-json-experiment/json) module.
+
+If this proposal is accepted, the implementation in `github.com/go-json-experiment/json`
+will be moved into the standard library.
+
+We may also provide a `golang.org/x/json` module that contains an identical copy
+of the implementation so that users on older Go releases can make use of v2.
+This module will use type-aliases to the Go standard library if the user
+is compiling with a sufficiently new version of the Go toolchain.
```

Comment:
ooh, clever. that means users won't have to wait for even one stable release of Go to happen with json/v2.

Reply:
Yep, the `google.golang.org/protobuf` module tends be held back quite a bit on Go releases, this is for the benefit of repos like those that need to support many older Go versions.